psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -2,130 +2,140 @@
-    private static final String SERVICE_PROVIDER = "SERVICE-PROVIDER";
+@CommandLine.Command(name = "psi-client",
+        mixinStandardHelpOptions = true,
+        description = "Simple client that communicates with the PSIEnclave using the web host.")
+public class Client implements Callable<Void> {
+
+    @CommandLine.Option(names = {"-r", "--role"}
+            , description = "Role Options: ${MERCHANT/SERVICE_PROVIDER}")
+    Role role = null;
+
+    //use picocli to provide command line parameters
+    @CommandLine.Parameters( description = "List of credit card numbers to be sent to enclave")
+        private List<String> creditCardNumbers = new ArrayList<String>();
+
+    @CommandLine.Option(names = {"-u", "--url"},
+                required = true,
+                description = "URL of the web host running the enclave.")
+        private String url;
+
+    @CommandLine.Option(names = {"-c", "--constraint"},
+                required = true,
+                description = "Enclave constraint which determines the enclave's identity and whether it's acceptable to use.",
+                converter = EnclaveConstraintConverter.class)
+    private EnclaveConstraint constraint;
 
-    public static void main(String[] args) throws InterruptedException, IOException, InvalidEnclaveException {
+    /**
+     * Use Kryo to serialize inputs from client to enclave
+     */
+    private Output serializeMessage(InputData listOfCreditCardNumbers){
+        Kryo kryo = new Kryo();
+        Output output = new Output(new ByteArrayOutputStream());
+        kryo.register(InputData.class, new InputDataSerializer());
+        kryo.writeObject(output, listOfCreditCardNumbers);
+        output.close();
+        return output;
+    }
 
-        if (args.length == 0) {
-            System.err.println("Please pass [MERCHANT/SERVICE-PROVIDER] [CONSTRAINT] followed by list of credit card numbers separated by spaces");
-            return;
-        }
-        String clientType = args[0];
-        String constraint = args[1];
-
-        //connect to host server
-        DataInputStream fromHost;
-        DataOutputStream toHost;
-        while (true) {
-            try {
-                System.out.println("Attempting to connect to localhost:9999");
-                Socket socket = new Socket();
-                socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), 9999), 5000);
-                fromHost = new DataInputStream(socket.getInputStream());
-                toHost = new DataOutputStream(socket.getOutputStream());
-                break;
-            } catch (Exception e) {
-                System.err.println("Retrying: " + e.getMessage());
-                Thread.sleep(2000);
+    enum Role {
+        MERCHANT,
+        SERVICE_PROVIDER
+    }
+
+    @Override
+    public Void call() throws Exception {
+        //a new private key is generated. Enclave Client is created using this private key and constraint.
```

Comment:
Comments should always begin with a capital letter.

Reply:
done